quinn-proto/src/endpoint.rs

```diff
@@ -1167,6 +928,11 @@ impl IndexMut<ConnectionHandle> for Slab<ConnectionMeta> {
     }
 }
 
+pub enum EndpointAppEvent {
```

Comment:
Not a huge fan of the name; this isn't exactly an application-facing enum. Maybe this?
```suggestion
pub enum DatagramEvent {
```
Or something else.